write-fonts/src/tables/glyf.rs

```diff
@@ -939,6 +1170,469 @@ mod tests {
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::off_curve(0, 1),
+                // CurvePoint::on_curve(1, 1),  // implied
+                CurvePoint::off_curve(2, 1),
+                CurvePoint::on_curve(2, 0),
+            ]],
+        );
+
+        let mut path2 = BezPath::new();
+        path2.move_to((0.0, 0.0));
+        path2.quad_to((0.0, 2.0), (2.0, 2.0)); // on-curve NOT equidistant from prev/next off-curves
+        path2.quad_to((3.0, 2.0), (3.0, 0.0));
+        path2.line_to((0.0, 0.0));
+        path2.close_path();
+
+        let glyphs = simple_glyphs_from_kurbo(&[path1, path2]).unwrap();
+
+        assert_contour_points(
+            &glyphs[0],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::off_curve(0, 1),
+                CurvePoint::on_curve(1, 1), // NOT implied
+                CurvePoint::off_curve(2, 1),
+                CurvePoint::on_curve(2, 0),
+            ]],
+        );
+        assert_contour_points(
+            &glyphs[1],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::off_curve(0, 2),
+                CurvePoint::on_curve(2, 2), // NOT implied
+                CurvePoint::off_curve(3, 2),
+                CurvePoint::on_curve(3, 0),
+            ]],
+        );
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_2_lines_open() {
+        let paths = make_interpolatable_paths(2, "MLL", false);
+        let glyphs = simple_glyphs_from_kurbo(&paths).unwrap();
+
+        assert_contour_points(
+            &glyphs[0],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::on_curve(1, 1),
+                CurvePoint::on_curve(2, 2),
+            ]],
+        );
+        assert_contour_points(
+            &glyphs[1],
+            vec![vec![
+                CurvePoint::on_curve(3, 3),
+                CurvePoint::on_curve(4, 4),
+                CurvePoint::on_curve(5, 5),
+            ]],
+        );
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_3_lines_open_duplicate_last_pt() {
+        let paths = make_interpolatable_paths(2, "MLLL", true);
+        let glyphs = simple_glyphs_from_kurbo(&paths).unwrap();
+
+        assert_contour_points(
+            &glyphs[0],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::on_curve(1, 1),
+                CurvePoint::on_curve(2, 2),
+                CurvePoint::on_curve(0, 0),
+            ]],
+        );
+        assert_contour_points(
+            &glyphs[1],
+            vec![vec![
+                CurvePoint::on_curve(3, 3),
+                CurvePoint::on_curve(4, 4),
+                CurvePoint::on_curve(5, 5),
+                CurvePoint::on_curve(3, 3),
+            ]],
+        );
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_4_lines_closed_duplicate_last_pt() {
+        for implicit_closing_line in &[true, false] {
```

Comment:
Optional, could we use one of the libraries you were exploring for parameterized tests here?

Reply:
yeah, but maybe in a different PR